lib/Graph/Graph.cpp

```diff
@@ -1605,6 +1605,30 @@ SaveNode *Function::createSave(Context &ctx, llvm::StringRef name,
   return addNode(new SaveNode(name, input, dest));
 }
 
+Node *Function::createDotProduct(llvm::StringRef name, NodeValue X,
+                                 NodeValue Y) {
+  auto XDimsSize = X.dims().size();
+
+  // Avoid unused variable warning in release builds.
+  (void)XDimsSize;
+
+  // X and Y must have the same shape.
+  assert(X.dims() == Y.dims() && "X and Y must have the same shape");
+  // X and Y must be 1D or 2D.
+  assert((XDimsSize == 1) || (XDimsSize == 2));
```

Comment:
Same here. Please move comment inside assert. And I think we can now delete comments, since their content is duplicated in the error message.